Makefile

```
@@ -585,8 +587,8 @@ endif
 ifeq "$(INSTALL_OCAMLNAT)" "true"
 	  $(INSTALL_PROG) ocamlnat$(EXE) "$(INSTALL_BINDIR)"
 endif
-	cd "$(INSTALL_COMPLIBDIR)" && \
-	   $(RANLIB) ocamlcommon.$(A) ocamlbytecomp.$(A) ocamloptcomp.$(A)
+# Some versions of ranlib do not support multiple archives
+	for archive in $(ARCHIVES); do $(RANLIB) $(INSTALL_COMPLIBDIR)/$$archive.$(A); done
```

Apologies, I was being sloppy before with quoting in my suggestions:
```suggestion
	for archive in $(ARCHIVES); do $(RANLIB) "$(INSTALL_COMPLIBDIR)/$$archive.$(A)"; done
```